### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by emyrue**
In this pull request, I:
- Created the Classroom class
- Created the Book class
- Created the Rental class
- Associated Classroom with Student
- Associated Person with Book via rental

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (Deepakdanger) [inline classroom.rb:4]: - [x] There should be `attr_reader :students`. Kindly add this.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
```
Comment (Deepakdanger) [inline book.rb]: - [x] There should be `rentals` also present in attr_accessor. Kindly add it
`attr_accessor :title, :author, :rentals`

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date
```
Comment (Deepakdanger) [inline rental.rb]: - [x] There should be `attr_accessor :date, :book, :person`. KIndly add it.

File: person.rb [hunk 4]
```diff
@@ -20,6 +21,12 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    return if @rentals.include?(rental)
+
+    @rentals.push(rental)
+  end
```
Comment (Deepakdanger) [inline person.rb]: - [x] Here passing `rental ` to a function not b able to create rental, I think the proper code should be this. Kindly add it.
```ruby
 def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    return if @rentals.include?(rental)
+
+    @rentals.push(rental)
+  end
```
Comment (Deepakdanger) [inline book.rb]: - [x] Here passing `rental ` to a function not b able to create rental, I think the proper code should be this. Kindly add it.
```ruby
 def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @emyrue, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 3 above)
Comment (Whoistolu) [reply rental.rb]: Well resolved 👍🏽

File: person.rb (hunk 4 above)
Comment (Whoistolu) [reply person.rb]: Well resolved 👍🏽

File: book.rb (hunk 5 above)
Comment (Whoistolu) [reply book.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @emyrue 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**